Update zwave.py

MarkoMarjamaa opened the pull request:
**Description:**
Added configuration option use_nodeid_in_names, default value is true that is the old style of device names. 
If it is set to false, no node ids are added to device names. This works if user can be sure the user can be sure there's no duplicate device names and if there are, renamed devices with OZWCP. 
This allows persistent zwave device names even after a device is broken and replaced.
Default value is true, and it is the same as the old behavior. If you don't use this configuration option, everything works like before. 

https://community.home-assistant.io/t/persistent-zwave-names-even-after-replacing-sensor-or-readding-to-controller/3101

**Related issue (if applicable):** fixes #

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#816

**Example entry for `configuration.yaml` (if applicable):**

``` yaml

```

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
- [ x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If code communicates with devices, web services, or a:
- [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
- [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L16)).
- [ ] New dependencies are only imported inside functions that use them ([example](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L51)).
- [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
- [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
- [ x ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
- [ ] Tests have been added to verify that the new code works.

Added configuration option use_nodeid_in_names, default value is true that is the old style of device names. 
If it is set to false, no node ids are added to device names. This works if user can be sure the user can be sure there's no duplicate device names and if there are, renanamed devices with OZWCP. 
This allows persistant zwave device names even after a device is broken and replaced.

MarkoMarjamaa committed:
Update zwave.py

Added configuration option use_nodeid_in_names, default value is true that is the old style of device names. 
If it is set to false, no node ids are added to device names. This works if user can be sure the user can be sure there's no duplicate device names and if there are, renanamed devices with OZWCP. 
This allows persistant zwave device names even after a device is broken and replaced.

balloob commented:
I don't like this change. Entity IDs are not user facing and thus should not be configurable. It will only be able to cause more confusion for our users.

MarkoMarjamaa commented:
"Entity IDs are not user facing and thus should not be configurable. It will only be able to cause more confusion for our users."

Entity IDs are already user configurable because they take the user configurable device name (if it's configured). Only thing that is removed is the number from the end of name. Normally the node id's are ok if the user is noob but if you need to save the sensor values for years and analyze the data even if the device node id's change, then the node id's have to removed from the device names.

balloob commented:
We moved to having node ids in the entity ids because people were getting entity id conflicts causing non-deterministic behavior in assigning entity ids when Home Assistant was restarted.

Assigning a switch for all devices is going to lead us down a rabbit hole. Next thing is that people want to be able to switch it per node id. And before we know it we will end up with a bunch of configuration options to serve the use case of a tiny percentage of the users.

If your data storage is the driving factor here, why not use an `UPDATE` query once a broken device has been replaced to make sure that history of device A is now seen as the history of device B. 

CC @turbokongen @robbiet480

turbokongen commented:
I agree. This is not wanted behaviour. Like @balloob says,  it's better to handle this in the database than replacing node ids in the core. They should be unique for the node. Take rfxtrx for example, there the packetid of the node is also unique. And cannot be changed. It's such a small use case for this, I would have to agree with @balloob on this.

MarkoMarjamaa commented:
If user does nothing, the switch defaults to true and everything works as before. Normal user does not need to know about this switch. 

I don't see why people would want this to be per node id, because if you want to have a number in the end of name, you can simply enter that number in OZWCP? 

Yes, I could of course update all my old data when the time comes, but the same could be achieved with a simply switch and ten lines of code.

MarkoMarjamaa committed:
Update zwave.py

Earlier version got temperature unit from (current) Temperature value. 
Danfoss Z Thermostat 014G0013 does not have that value, only setpoint temperature.
Adding Danfoss ended in error unit_of_measure being None.
Now unit is also now updated from setpoint temperature value. 
Is there  thermostat that has no setpoint temperature?

MarkoMarjamaa committed:
Update zwave.py

Rolled back earlier change

turbokongen commented:
I completely forgot, but zwave support replacing failed nodes for a id, so this is not really needed at  all😉
